docs/samples/git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v4-Y-Z-1-no-hwtemplate.yaml

```diff
@@ -0,0 +1,257 @@
+              up to two for dual-stack clusters (at most one IP address per IP
+              stack used). The order of stacks should be the same as order of
+              subnets in Cluster Networks, Service Networks, and Machine Networks.
+            items:
+              type: string
+            maxItems: 2
+            type: array
+          baseDomain:
+            description: BaseDomain is the base domain to use for the deployed
+              cluster.
+            type: string
+          clusterName:
+            description: ClusterName is the name of the cluster.
+            type: string
+          extraAnnotations:
+            additionalProperties:
+              additionalProperties:
+                type: string
+              type: object
+            description: Additional cluster-wide annotations to be applied to
+              the rendered templates
+            type: object
+          extraLabels:
+            additionalProperties:
+              additionalProperties:
+                type: string
+              type: object
+            description: Additional cluster-wide labels to be applied to the rendered
+              templates
+            type: object
+          ingressVIPs:
+            description: IngressVIPs are the virtual IPs used for cluster ingress
+              traffic. Enter one IP address for single-stack clusters, or up to
+              two for dual-stack clusters (at most one IP address per IP stack
+              used). The order of stacks should be the same as order of subnets
+              in Cluster Networks, Service Networks, and Machine Networks.
+            items:
+              type: string
+            maxItems: 2
+            type: array
+          machineNetwork:
+            description: MachineNetwork is the list of IP address pools for machines.
+            items:
+              description: MachineNetworkEntry is a single IP address block for
+                node IP blocks.
+              properties:
+                cidr:
+                  description: CIDR is the IP block address pool for machines
+                    within the cluster.
+                  type: string
+              required:
+              - cidr
+              type: object
+            type: array
+          nodes:
+            items:
+              description: NodeSpec
+              properties:
+                bmcAddress:
+                  description: BmcAddress holds the URL for accessing the controller
+                    on the network.
+                  type: string
+                bmcCredentialsName:
+                  description: BmcCredentialsName is the name of the secret containing
+                    the BMC credentials (requires keys "username" and "password").
+                  properties:
+                    name:
+                      type: string
+                  required:
+                  - name
+                  type: object
```

Comment:
I think we're requiring the ProvisioningRequest to have the username/password - why do we need the secret here as well? Ahh... this will be used to provide a name for the created secret. Hmm... could we auto-generate the secret name (e.g. based on the cluster name and node name)? Would be simpler for the user and avoid errors where the user attempts to use the same secret name for multiple nodes.

Reply:
The ProvisioningRequest wouldn't have the BMC username/password, afaik. That would ultimately come from the hardware manager when nodes are allocated, and the plugin would create a corresponding secret

Reply:
In this use case, there is no H/W manager.

Reply:
`bmcCredentialsName` is optional field. Yes, currently, we auto-generate the secret name as <nodename>-bmc-secret if secret name is not provided. Right, it seems unnecessary since we’re auto-generating it anyway. I don't see a reason why users would need a different secret name 🤔 .

Reply:
Excluded `bmcCredentialsName` parameter from schema.